test/Marketplace.js

```diff
@@ -70,199 +89,276 @@ contract('Marketplace', function([_, owner, seller, buyer]) {
-    // Event emitted
-    logs.length.should.be.equal(1)
-    checkAuctionCreatedLog(logs[0], assetId, seller, newPrice, newEndTime)
+    it('should fail to create an order :: (address not the owner of asset)', async function() {
+      await market
+        .createOrder(assetId, erc721.address, itemPrice, endTime, {
+          from: otherAddress
+        })
+        .should.be.rejectedWith(EVMRevert)
+    })
 
-    // Check data
-    let s = await market.auctionByAssetId(assetId)
-    s[1].should.be.equal(seller)
-    s[2].should.be.bignumber.equal(newPrice)
-    s[3].should.be.bignumber.equal(newEndTime)
+    it('should fail to create an order :: (not an ERC721 contract)', async function() {
+      await market
+        .createOrder(assetId, erc20.address, itemPrice, endTime, {
+          from: seller
+        })
+        .should.be.rejectedWith(EVMRevert)
+    })
   })
 
-  // Cancel
-
-  it('should cancel a created order', async function() {
-    let itemPrice = web3.toWei(1.0, 'ether')
-
-    await market.createOrder(assetId, itemPrice, endTime, {
-      from: seller
+  describe('Cancel', function() {
+    it('should cancel a created order', async function() {
+      await market.createOrder(assetId, erc721.address, itemPrice, endTime, {
+        from: seller
+      })
+      const { logs } = await market.cancelOrder(erc721.address, assetId, {
+        from: seller
+      })
+
+      // Event emitted
+      logs.length.should.be.equal(1)
+      checkOrderCancelledLog(logs[0], assetId, seller, erc721.address)
     })
-    const { logs } = await market.cancelOrder(assetId, { from: seller })
-
-    // Event emitted
-    logs.length.should.be.equal(1)
-    checkAuctionCancelledLog(logs[0], assetId, seller)
-  })
 
-  it('should fail canceling an order', async function() {
-    let itemPrice = web3.toWei(1.0, 'ether')
+    it('should fail canceling an order :: (wrong user)', async function() {
+      await market.createOrder(assetId, erc721.address, itemPrice, endTime, {
+        from: seller
+      })
+      await market
+        .cancelOrder(erc721.address, assetId, { from: buyer })
+        .should.be.rejectedWith(EVMRevert)
+    })
 
-    await market.createOrder(assetId, itemPrice, endTime, {
-      from: seller
+    it('should fail canceling an order :: (wrong address)', async function() {
+      await market.createOrder(assetId, erc721.address, itemPrice, endTime, {
+        from: seller
+      })
+      await market
+        .cancelOrder(erc20.address, assetId, { from: seller })
+        .should.be.rejectedWith(EVMRevert)
     })
-    await market
-      .cancelOrder(assetId, { from: buyer })
-      .should.be.rejectedWith(EVMRevert)
   })
 
-  // Execute
-
-  it('should execute a created order', async function() {
-    let itemPrice = web3.toWei(1.0, 'ether')
-
-    await market.createOrder(assetId, itemPrice, endTime, {
-      from: seller
+  describe('Execute', function() {
+    it('should execute a created order', async function() {
```

Comment:
Maybe we could write a test to Double Execute an order. This will test that we are correctly removing the Order from the mapping.